test/integration/local/enrich-payment-request.test.js

```diff
@@ -1,5 +1,8 @@
+const { GBP } = require('../../../app/currency')
 const db = require('../../../app/data')
 const enrichPaymentRequest = require('../../../app/enrichment')
+const { AP } = require('../../../app/ledgers')
+const { M12 } = require('../../../app/schedules')
 let scheme
 let schemeCode
 let paymentRequest
```

Comment:
Just a few specifics missing I feel, `should error for empty payment request` is fine but I'd also want to have 2 more tests for 'What kind of Error was thrown?' and (more importantly) 'What error message was thrown?' -- there are a lot of function call outs for `enrichPaymentRequest` and I'm not sure which function would trip up first w/ an empty object

What if `enrichPaymentRequest` has inputs of
- undefined
- null
- []
- ''

Reply:
This is a good point, I think I'd taken for granted that these tests were already in a good state.  But you're absolutely right they can be improved and I believe it falls in scope of this ticket.

I've added the tests you suggest.  I did consider adding specifics for different states of the payment request object, but they're already covered by the validation and schema tests.

For the error message, there's two category of message, either they will result in a validation failure or a processing error as is.

Now if a validation error is thrown here the message is rejected and the source system is informed they've sent us something bad and that's the end of the journey.

But if it's a general error then the message will be abandoned, but picked up again after the lock period expires to try again until the max 2000 attempts is reached.  But in these scenarios we will know that we'll never be able to successfully process the message as the content is wrong.  So we should be immediately rejecting these. 

So I will update so they all return as validation errors and include that in the tests.

Reply:
I have now added this check and they all successfully immediately return as validation errors as expected.